pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -430,7 +431,8 @@ protected boolean consumeLoop()
       // Update _currentOffset upon return from this method
       MessageBatch messageBatch;
       try {
-        messageBatch = _partitionGroupConsumer.fetchMessages(_currentOffset, _streamConfig.getFetchTimeoutMillis());
+        messageBatch = _partitionGroupConsumer.fetchMessages(_currentOffset, _segmentZKMetadata.getContinuationMode(),
```

Comment:
(MAJOR) Only the first batch should be count as ingested from new stream.
Within the `fetchMessages()` API, we should just a boolean `inclusive` to mark whether we should consume the current offset